### DISCUSSION TITLE
Vacuum mode names reset every reboot.

### DISCUSSION THREAD

**Discussion by dimkagithub** [merdok/homebridge-miot#525] [General]
Hi!

Vacuum is **Dreame D9**.
If Homebridge server is rebooted all vacuum mode names resets to its default names (Mode - Basic, Mode - Full Speed, etc). After each reboot I have to rename it back to my language.
This happens only in Home App only. In Homebridge Web Interface they don't reset.
What can be wrong?
Hope it is clear.

**Comment by merdok** [comment]
Hi,

yeah i am aware of this issue. It is currently due to a workaround for the switch names as before as the switches had the same name. I will try to fix that as soon as i have some time...

**Reply by dimkagithub** [reply]
Hi!
Not fixed yet?

**Reply by merdok** [reply]
Not yet...

**Reply by dimkagithub** [reply]
Hi!
Not fixed yet? :)

**Reply by merdok** [reply]
Sorry, did not have time to deeply look into that, possibly would require some big changes...
